include/boost/uuid/sha1.hpp

```diff
@@ -1,231 +1,16 @@
-// boost/uuid/sha1.hpp header file  ----------------------------------------------//
+/*
+ * Distributed under the Boost Software License, Version 1.0.
+ *    (See accompanying file LICENSE_1_0.txt or copy at
+ *          http://www.boost.org/LICENSE_1_0.txt)
+ */
 
-// Copyright 2007 Andy Tompkins.
-// Distributed under the Boost Software License, Version 1.0. (See
-// accompanying file LICENSE_1_0.txt or copy at
-// http://www.boost.org/LICENSE_1_0.txt)
+#ifndef BOOST_UUID_SHA1_DEPRECATED_HPP
+#define BOOST_UUID_SHA1_DEPRECATED_HPP
 
-// Revision History
-//  29 May 2007 - Initial Revision
-//  25 Feb 2008 - moved to namespace boost::uuids::detail
-//  10 Jan 2012 - can now handle the full size of messages (2^64 - 1 bits)
+#include <boost/uuid/detail/sha1.hpp>
 
-// This is a byte oriented implementation
-
-#ifndef BOOST_UUID_SHA1_H
-#define BOOST_UUID_SHA1_H
-
-#include <boost/static_assert.hpp>
-#include <stdexcept>
-#include <boost/throw_exception.hpp>
-#include <cstddef>
-#include <string>
-
-#ifdef BOOST_NO_STDC_NAMESPACE
-namespace std {
-    using ::size_t;
-} // namespace std
+#if defined(__GNUC__) || defined(_MSC_VER)
+#pragma message("This header is deprecated, use boost/uuid/detail/sha1.hpp instead.")
```

Comment:
I think, the message should say that the header is implementation detail.

Reply:
Given sha1 and seed_rng are already in the detail namespace, am I required to provide this mechanism?  I believe the answer is no, but I want to make the transition smooth because I suspect there are many projects that rely on uuid sha1 just for the sha1 implementation.

Reply:
I don't think relying on Boost.UUID just for the SHA1 implementation is reasonable. Especially given that it is an implementation detail. That said, the header is in the public directory, so I still would give it a deprecation period. My comment was just about the message, which currently suggests to continue using the header at the new location while I think it should make clear that people shouldn't be using it in the first place.

Reply:
I found that,

Boost.Compute is including uuid/sha1: ``compute/include/boost/compute/detail/sha1.hpp``
Boost.Beast has another sha1 implementation: ``beast/include/boost/beast/core/detail/sha1.hpp``

Usually when there is more than one available, it's time to start looking into consolidating.  This is why I asked the question about something like Boost.Hash or Boost.Crypto to hold commonly used things like this.  The pushback was that area usually has a lot of specialized functions and it would be better to leave it out.  So for now I guess we have two SHA1 implementations.  Boost.Compute technically needs one of its own so it does not rely on uuid/detail, or we make a more common library for it.